File: 프로그래머스/1/42748. K번째수/K번째수.py
def solution(array, commands):
    answer = []
    for c in commands : 
        i, j, k = c[0], c[1], c[2]
        newArray = sorted(array[i-1:j])
        if len(newArray) > k :
            answer.append(newArray[k-1])
        else: 
            answer.append(newArray[-1])
    return answer
# i = 배열의 시작, j = 배열의 끝, k = 뽑은 배열을 정렬 후 해당 번째 

File: 프로그래머스/1/42748. K번째수/K번째수.java
import java.util.*;

/*
array
command => command.length동안 수행
command [i,j,k] => i번째부터 j까지 잘라서 정렬 후 k번째 숫자 리턴하여 결과 배열 추가
*/
class Solution {
        public int[] solution(int[] array, int[][] commands) {
            int[] answer = {};
            answer = new int[commands.length];

            for (int q=0; q<commands.length; q++){
                int[] singleCmd = commands[q];
                int i = singleCmd[0]; int j = singleCmd[1]; int k = singleCmd[2];

                // 인덱스로 copy slice
                int[] resList = Arrays.copyOfRange(array, i-1, j);
                Arrays.sort(resList);
                answer[q] = resList[k-1];
            }
            return answer;
        }
    }